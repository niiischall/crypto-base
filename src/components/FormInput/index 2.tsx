import React from 'react';
import { get } from 'lodash';
import { TextField, withStyles } from '@material-ui/core';

interface Props {
  input: any;
  meta: any;
  [x: string]: any;
}

export const StyledTextField = withStyles((theme) => ({
  root: {
    height: 50,
    width: '100%',
    margin: '17.5px 0px',
    '& label.Mui-focused': {
      color: 'green',
    },
    '& .MuiInput-underline:after': {
      borderBottomColor: '#669900',
    },
    '& .MuiOutlinedInput-root': {
      '& fieldset': {
        borderColor: '#669900',
        borderWidth: '2px',
      },
      '&:hover fieldset': {
        borderColor: '#f59842',
        borderWidth: '2px',
      },
      '&.Mui-focused fieldset': {
        borderColor: '#669900',
        borderWidth: '2px',
      },
    },
    '& .Mui-error': {
      fontSize: '10px',
    },
  },
}))((props: any) => <TextField {...props} />);

const TextFieldInput: React.FC<Props> = (props) => (
  <StyledTextField
    error={props.meta.touched && props.meta.error}
    helperText={props.meta.touched && props.meta.error}
    InputProps={{
      ...props.InputProps,
      style: { fontSize: 14 },
    }}
    InputLabelProps={{
      style: { fontSize: 14, color: '#6d758d' },
    }}
    onChange={(event: any, index: any, value: any) =>
      props.input.onChange(value)
    }
    value={get(props, 'input.value', '')}
    variant="outlined"
    {...props.input}
    {...props}
  />
);

export default TextFieldInput;
