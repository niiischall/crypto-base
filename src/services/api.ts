const baseCoinAPI = 'https://pro-api.coinmarketcap.com/';
const baseAuthAPI = 'https://identitytoolkit.googleapis.com/';
const apiVersion = 'v1';
const authVersion = 'v1';
const apiKey = `&CMC_PRO_API_KEY=${process.env.REACT_APP_CMC_KEY}`;
const authKey = `?key=${process.env.REACT_APP_AUTH_KEY}`;

export const apiEndPoints = {
  totalCoins:
    '/cryptocurrency/map?sort=cmc_rank&limit=500&listing_status=active',
  popularCoins: '/cryptocurrency/listings/latest?limit=25',
  trendingCoins:
    '/cryptocurrency/listings/latest?sort=percent_change_24h&limit=25',
  coinsInfo: '/cryptocurrency/info?slug=',
  coinsQuotes: '/cryptocurrency/quotes/latest?slug=',
  signUp: `/accounts:signUp`,
  signIn: `/accounts:signInWithPassword`,
};

export async function fetchApi(
  endPoint: string,
  payload?: {} | undefined,
  method: string = 'get',
  headers?: {},
) {
  let path = `${baseCoinAPI}${apiVersion}${endPoint}${apiKey}`;
  const headersObject: any = {
    ...headers,
  };
  let request = {
    body: JSON.stringify(payload),
    headers: headersObject,
    method: method.toLowerCase(),
  };

  try {
    return fetch(path, request);
  } catch (e: any) {
    const stringError = e && e.toString && e.toString();
    const type =
      stringError === 'TypeError: Network request failed'
        ? 'networkError'
        : 'unknown';
    const error = {
      text: stringError,
      type,
    };
    throw error;
  }
}

export async function authApi(
  endPoint: string,
  method: string,
  payload: {},
  headers?: {},
) {
  const path = `${baseAuthAPI}${authVersion}${endPoint}${authKey}`;
  const headersObject: any = {
    ...headers,
  };
  let request = {
    body: JSON.stringify(payload),
    headers: headersObject,
    method: method.toLowerCase(),
  };

  try {
    return fetch(path, request);
  } catch (e: any) {
    const stringError = e && e.toString && e.toString();
    const type =
      stringError === 'TypeError: Network request failed'
        ? 'networkError'
        : 'unknown';
    const error = {
      text: stringError,
      type,
    };
    throw error;
  }
}
