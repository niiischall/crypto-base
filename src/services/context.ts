import React from 'react';

//Page Values
export const pages = {
  home: 'home',
  search: 'search',
  profile: 'profile',
};

//Page context
const pageContext = React.createContext({
  page: '',
  switchPage: (event: any, value: string) => {},
});

//Page Provider
export const PageProvider = pageContext.Provider;

//Default pageContext Export
export default pageContext;
