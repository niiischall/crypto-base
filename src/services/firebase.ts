import { initializeApp } from 'firebase/app';
import { getFirestore } from 'firebase/firestore';

const firebaseConfig = {
  apiKey: 'AIzaSyD72gMheiysnrSBbCaNNjNqfkFDw5zyWgY',
  authDomain: 'crypto-base-db.firebaseapp.com',
  databaseURL:
    'https://crypto-base-db-default-rtdb.asia-southeast1.firebasedatabase.app',
  projectId: 'crypto-base-db',
  storageBucket: 'crypto-base-db.appspot.com',
  messagingSenderId: '853554508159',
  appId: '1:853554508159:web:9a2f30558c2fc026066a4a',
};

initializeApp(firebaseConfig);
export default getFirestore();
