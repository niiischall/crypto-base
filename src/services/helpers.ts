export const API_RECALL_TIMER = 900000;

export function normalizePrice(value: any, precision: any = 0) {
  let toReturn = value;
  if (!value) {
    return null;
  }
  const min = 0;
  const max = 99999999;
  if (typeof toReturn === 'string') {
    toReturn = toReturn.replace(/[^0-9.]+/g, '');
    toReturn = parseFloat(toReturn);
  }

  if (toReturn < min) {
    toReturn = min;
  }
  if (toReturn > max) {
    toReturn = max;
  }
  const formatted = toReturn
    .toFixed(precision)
    .replace(/./g, (c: any, i: any, a: any) =>
      i && c !== '.' && (a.length - i) % 3 === 0 ? `,${c}` : c,
    );
  return formatted;
}

export function isCoinPresent(coin: any, data: any[]) {
  return data.find((element: any) => element.slug === coin.slug);
}

export function normalizeText(value: any) {
  if (!value) {
    return null;
  }
  return value;
}

export function normalizeError(error: string) {
  let normalizedError = error;
  normalizedError = error.replace(/_/g, ' ');
  return normalizedError.toLowerCase();
}
