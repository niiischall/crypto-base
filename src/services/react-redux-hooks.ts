import {
  useSelector as originalUseSelector,
  useDispatch as originalUseDispatch,
} from 'react-redux';

export const useSelector: any = (state: any) => originalUseSelector(state);
export const useDispatch: any = () => originalUseDispatch();
