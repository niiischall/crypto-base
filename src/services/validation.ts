import { email } from '@sideway/address';

//Validations
const minLength = (min: number) => (value: any) =>
  value && value.length < min ? `Must be ${min} characters or more` : undefined;

export const validations = {
  required: (value: any) => (value ? undefined : 'Required'),
  minLength8: minLength(8),
  email: (value: string) =>
    value && !email.isValid(value) ? 'Invalid email address' : undefined,
};

export const setUpValidation =
  (...validators: any) =>
  (value: any) =>
    validators.reduce(
      (error: any, validator: any) => error || validator(value),
      undefined,
    );
