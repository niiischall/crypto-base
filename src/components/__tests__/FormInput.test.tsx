import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { store } from '../../mocks/mock';
import * as ReactReduxHooks from '../../services/react-redux-hooks';

import FormInput from '../FormInput';

const props = {
  input: {},
  meta: {},
  x: [],
};

const setup = () => {
  const component = mount(
    <Provider store={store}>
      <FormInput {...props} />
    </Provider>,
  );
  return {
    component,
  };
};

describe('FormInput Component', () => {
  let enzymeWrapper: any;
  const { component } = setup();

  beforeEach(() => {
    /* mocking useSelector on our mock store */
    jest
      .spyOn(ReactReduxHooks, 'useSelector')
      .mockImplementation((state) => store.getState());
    /* mocking useDispatch on our mock store  */
    jest
      .spyOn(ReactReduxHooks, 'useDispatch')
      .mockImplementation(() => store.dispatch);

    /* shallow rendering */
    enzymeWrapper = component;
  });

  it('should match snapshot', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
