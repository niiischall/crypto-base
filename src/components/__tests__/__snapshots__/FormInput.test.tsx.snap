// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Component should match snapshot 1`] = `ReactWrapper {}`;
